src/main/java/mate/academy/security/AuthenticationServiceImpl.java

```diff
@@ -0,0 +1,36 @@
+package mate.academy.security;
+
+import java.util.Optional;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.exception.RegistrationException;
+import mate.academy.lib.Inject;
+import mate.academy.lib.Service;
+import mate.academy.model.User;
+import mate.academy.service.UserService;
+import mate.academy.util.HashUtil;
+
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String email, String password) throws AuthenticationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        User user = userOptional.get();
+        String hashPassword = HashUtil.hashPassword(password, user.getSalt());
+        if (!userOptional.isEmpty() && user.getPassword().equals(hashPassword)) {
+            return user;
+        }
+        throw new AuthenticationException("Can`t authenticate user");
+    }
+
+    @Override
+    public User register(String email, String password) throws RegistrationException {
+        Optional<User> userOptional = userService.findByEmail(email);
+        if (userOptional.isEmpty()) {
+            throw new RegistrationException("Can`t registration user");
+        }
+        return userService.add(new User(email, password));
```

Comment:
does it work correctly, your code throws an exception when you to register **a new  user** and his email is **not present** in the db